### PR TITLE
get experience by index & delete experience by index

### PR DESCRIPTION
**Pull request by lh5844**
resolves issue #19 and issue #4

tests fail due to issue #15 not being resolved yet

### CODE REVIEW

**Comment by andyjianzhou** [comment]
Hey, this conflicts with #28. I fixed testcases and the get and post experiences implemented.

**Comment by Pradyuman7** [comment]
> Hey, this conflicts with #28. I fixed testcases and the get and post experiences implemented. Not the delete functionality though. I implemented it so that it saves and generates IDs, which helps you with #4

We can wait to merge that PR first then. What do you think @lh5844?

**Comment by Pradyuman7** [comment]
That PR has been merged, you can make the required changes now @lh5844

**Comment by IvanKolchanov** [comment]
Oh, I was looking at this code because I was implementing DELETE for /resume/skill and I feel like there is bug in the code. After deleting the experience on line 66 in app.py, you need to use "save_data('data/data.json', data)" to save the data, otherwise, there will be no change to the data.json file. And you should probably add checking the data to the test case too, so that it doesn't miss something like that. :-)

**Comment by lh5844** [comment]
Thank you for the catch! I made the changes to delete request for experience

**Comment by lh5844** [comment]
@Pradyuman7 this is ready for review

File: app.py [hunk 1]
```diff
@@ -43,11 +43,38 @@ def experience():
...
         return jsonify({'id': new_id}), 201
     
+    if request.method == 'DELETE':
+        index = request.args.get("index")
+        # check index is valid number
+        if index is not None:
+            if not index.isnumeric():
+                return jsonify({"error": "Invalid Index"}), 400
+            
+        if 0 < int(index) <= len(data["experience"]):
```
Comment (Pradyuman7) [inline app.py:55]: Don't we reach this branch with index = None. This might cause problems here

File: app.py [hunk 2]
```diff
@@ -43,11 +43,38 @@ def experience():
...
+            data["experience"].pop(int(index)-1)
+            save_data('data/data.json', data)
+            return jsonify({"message": "Successfully deleted"}), 200
+
+        return jsonify({"error": 'Invalid Index'}), 400
+
     return jsonify({'error': 'Method not allowed'}), 405
 
+
+@app.route('/resume/experience/<int:index>', methods = ['GET'])
```
Comment (Pradyuman7) [inline app.py:66]: iirc we're not using index as a url param, but similar to what you have in the DELETE endpoint

**Comment by lh5844** [comment]
@Pradyuman7 Just made the changes, thank you for pointing them out

**Review by Pradyuman7 — APPROVED** [review]
LGTM 👍

**Pull request merged**